server: TestCheckRestartSafe_AllowMinimumQuorum_Pass failed

server.TestCheckRestartSafe_AllowMinimumQuorum_Pass [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20189851?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20189851?buildTab=artifacts#/) on master @ [da4cfcb2b4840ae377cd2b158a67a5da686b118f](https://github.com/cockroachdb/cockroach/commits/da4cfcb2b4840ae377cd2b158a67a5da686b118f):


```
=== RUN   TestCheckRestartSafe_AllowMinimumQuorum_Pass
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/077c6015fca14174d23f0b07262a46e6/logTestCheckRestartSafe_AllowMinimumQuorum_Pass550858904
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    api_v2_test.go:455: 
        	Error Trace:	pkg/server/api_v2_test.go:455
        	Error:      	Should be true
        	Test:       	TestCheckRestartSafe_AllowMinimumQuorum_Pass
    panic.go:635: -- test log scope end --
--- FAIL: TestCheckRestartSafe_AllowMinimumQuorum_Pass (19.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_AllowMinimumQuorum_Pass.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

server.TestCheckRestartSafe_AllowMinimumQuorum_Pass [failed](https://mesolite.cluster.engflow.com/invocations/default/85d063e4-ee96-4e9e-a31b-a9cacc111561?testReportRun=3&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [c0b904a99b6a067b8ca17f5a60e7e0d0f3d79883](https://github.com/cockroachdb/cockroach/commits/c0b904a99b6a067b8ca17f5a60e7e0d0f3d79883):


```
=== RUN   TestCheckRestartSafe_AllowMinimumQuorum_Pass
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass229782293
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    api_v2_test.go:455: 
        	Error Trace:	pkg/server/api_v2_test.go:455
        	Error:      	Should be true
        	Test:       	TestCheckRestartSafe_AllowMinimumQuorum_Pass
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass229782293
--- FAIL: TestCheckRestartSafe_AllowMinimumQuorum_Pass (31.48s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_AllowMinimumQuorum_Pass.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

server.TestCheckRestartSafe_AllowMinimumQuorum_Pass [failed](https://mesolite.cluster.engflow.com/invocations/default/f72e95c5-6810-4316-bebd-9aa43bd9a025?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [b8c2405930718b735bf009c802798b4918b66631](https://github.com/cockroachdb/cockroach/commits/b8c2405930718b735bf009c802798b4918b66631):

Fatal error:

```
panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc013a29770 0xc016a5ac00 {{} 1} {[] {} 0xc00c7ac600} {[] {} <nil>} {{{{0 0}}} 0xc013979200}})]}) [recovered]
	panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc013a29770 0xc016a5ac00 {{} 1} {[] {} 0xc00c7ac600} {[] {} <nil>} {{{{0 0}}} 0xc013979200}})]}) [recovered]
	panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc013a29770 0xc016a5ac00 {{} 1} {[] {} 0xc00c7ac600} {[] {} <nil>} {{{{0 0}}} 0xc013979200}})]}) [recovered]
	panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc013a29770 0xc016a5ac00 {{} 1} {[] {} 0xc00c7ac600} {[] {} <nil>} {{{{0 0}}} 0xc013979200}})]}) [recovered]
	panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc013a29770 0xc016a5ac00 {{} 1} {[] {} 0xc00c7ac600} {[] {} <nil>} {{{{0 0}}} 0xc013979200}})]})

```
Stack: 

```
goroutine 812769 [running]:
testing.tRunner.func1.2({0x604fca0, 0xc017b7f3f0})
	GOROOT/src/testing/testing.go:1632 +0x230
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1635 +0x35e
panic({0x604fca0?, 0xc017b7f3f0?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	pkg/util/leaktest/leaktest.go:136 +0x3aa
panic({0x604fca0?, 0xc017b7f3f0?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x474c45?, {0x73956f8, 0xb741300})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x604fca0?, 0xc017b7f3f0?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc002ecc990, {0x73956f8, 0xb741300})
	pkg/util/stop/stopper.go:566 +0x165
panic({0x604fca0?, 0xc017b7f3f0?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).SetDraining.func1(0xc013a29770, {0x7396140, 0xc00a52c780}, 0x1)
	pkg/sql/catalog/lease/lease.go:1601 +0x1db
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).SetDraining(0xc01107bb88, {0x7396140, 0xc00a52c780}, 0x1, 0x0, 0x1)
	pkg/sql/catalog/lease/lease.go:1604 +0x187
github.com/cockroachdb/cockroach/pkg/server.(*drainServer).drainClients(0xc012bad400, {0x73956f8?, 0xb741300?}, 0x0)
	pkg/server/drain.go:489 +0x4ce
github.com/cockroachdb/cockroach/pkg/server.(*testServer).DrainClients(0xc013304540?, {0x73956f8?, 0xb741300?})
	pkg/server/testserver.go:1943 +0x2d
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).DrainClients(0x73956f8?, {0x73956f8, 0xb741300})
	bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/app_forwarder_generated.go:326 +0x3f
github.com/cockroachdb/cockroach/pkg/server.drain({0x73956f8, 0xb741300}, {0x74a3728, 0xc013304540}, 0xc00b875d40)
	pkg/server/api_v2_test.go:541 +0xa6
github.com/cockroachdb/cockroach/pkg/server.TestCheckRestartSafe_AllowMinimumQuorum_Pass(0xc00b875d40)
	pkg/server/api_v2_test.go:449 +0x265
testing.tRunner(0xc00b875d40, 0x6a025e8)
	GOROOT/src/testing/testing.go:1690 +0xf4
created by testing.(*T).Run in goroutine 1
	GOROOT/src/testing/testing.go:1743 +0x390
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestCheckRestartSafe_AllowMinimumQuorum_Pass
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass1868956342
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    stopper.go:226: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass1868956342
    stopper.go:226: panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc013a29770 0xc016a5ac00 {{} 1} {[] {} 0xc00c7ac600} {[] {} <nil>} {{{{0 0}}} 0xc013979200}})]})
--- FAIL: TestCheckRestartSafe_AllowMinimumQuorum_Pass (32.06s)
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_AllowMinimumQuorum_Pass.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

server.TestCheckRestartSafe_AllowMinimumQuorum_Pass [failed](https://mesolite.cluster.engflow.com/invocations/default/7556b8a6-f540-445d-9d76-4c406d585bb4?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [b8c2405930718b735bf009c802798b4918b66631](https://github.com/cockroachdb/cockroach/commits/b8c2405930718b735bf009c802798b4918b66631):

Fatal error:

```
panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc0065de2a0 0xc0034ae600 {{} 1} {[] {} 0xc0081671c0} {[] {} <nil>} {{{0 0} 0} 0xc0061097a0}})]}) [recovered]
	panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc0065de2a0 0xc0034ae600 {{} 1} {[] {} 0xc0081671c0} {[] {} <nil>} {{{0 0} 0} 0xc0061097a0}})]}) [recovered]
	panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc0065de2a0 0xc0034ae600 {{} 1} {[] {} 0xc0081671c0} {[] {} <nil>} {{{0 0} 0} 0xc0061097a0}})]}) [recovered]
	panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc0065de2a0 0xc0034ae600 {{} 1} {[] {} 0xc0081671c0} {[] {} <nil>} {{{0 0} 0} 0xc0061097a0}})]}) [recovered]
	panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc0065de2a0 0xc0034ae600 {{} 1} {[] {} 0xc0081671c0} {[] {} <nil>} {{{0 0} 0} 0xc0061097a0}})]})

```
Stack: 

```
goroutine 5263 [running]:
testing.tRunner.func1.2({0x991da60, 0xc01dcd9000})
	GOROOT/src/testing/testing.go:1632 +0x3fc
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1635 +0x6b6
panic({0x991da60?, 0xc01dcd9000?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	pkg/util/leaktest/leaktest.go:136 +0x51a
panic({0x991da60?, 0xc01dcd9000?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc01fef6e80?, {0xb355cc8, 0x109ba080})
	pkg/util/stop/stopper.go:226 +0x74
panic({0x991da60?, 0xc01dcd9000?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc0047e9540, {0xb355cc8, 0x109ba080})
	pkg/util/stop/stopper.go:566 +0x2c5
panic({0x991da60?, 0xc01dcd9000?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).SetDraining.func1(0xc0065de2a0, {0xb356700, 0xc01dee4f00}, 0x1)
	pkg/sql/catalog/lease/lease.go:1601 +0x345
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).SetDraining(0xc001dc1608, {0xb356700, 0xc01dee4f00}, 0x1, 0x0, 0x1)
	pkg/sql/catalog/lease/lease.go:1604 +0x22a
github.com/cockroachdb/cockroach/pkg/server.(*drainServer).drainClients(0xc001092be0, {0xb355cc8, 0x109ba080}, 0x0)
	pkg/server/drain.go:489 +0xa4f
github.com/cockroachdb/cockroach/pkg/server.(*testServer).DrainClients(0xc006b00708, {0xb355cc8, 0x109ba080})
	pkg/server/testserver.go:1943 +0x85
github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).DrainClients(0xc0042cf3b0, {0xb355cc8, 0x109ba080})
	bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/app_forwarder_generated.go:326 +0x56
github.com/cockroachdb/cockroach/pkg/server.drain({0xb355cc8, 0x109ba080}, {0xb46e9e8, 0xc0042cf3b0}, 0xc000217d40)
	pkg/server/api_v2_test.go:541 +0xc2
github.com/cockroachdb/cockroach/pkg/server.TestCheckRestartSafe_AllowMinimumQuorum_Pass(0xc000217d40)
	pkg/server/api_v2_test.go:449 +0x37f
testing.tRunner(0xc000217d40, 0xa2dc3b0)
	GOROOT/src/testing/testing.go:1690 +0x227
created by testing.(*T).Run in goroutine 1
	GOROOT/src/testing/testing.go:1743 +0x826
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestCheckRestartSafe_AllowMinimumQuorum_Pass
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass208382668
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    stopper.go:226: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass208382668
    stopper.go:226: panic: descriptor leak was detected for: 1 ({[%!s(*lease.descriptorVersionState=&{0xc0065de2a0 0xc0034ae600 {{} 1} {[] {} 0xc0081671c0} {[] {} <nil>} {{{0 0} 0} 0xc0061097a0}})]})
--- FAIL: TestCheckRestartSafe_AllowMinimumQuorum_Pass (93.67s)
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_AllowMinimumQuorum_Pass.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/server/api_v2_test.go
+++ b/pkg/server/api_v2_test.go
@@ -338,7 +338,7 @@ func TestCheckRestartSafe_Criticality(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 150810)
+	skip.UnderDuress(t)
 
 	ctx := context.Background()
 
@@ -436,7 +436,7 @@ func TestCheckRestartSafe_AllowMinimumQuorum_Pass(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 150365)
+	skip.UnderDuress(t)
 
 	ctx := context.Background()
 	var err error
@@ -477,7 +477,7 @@ func TestCheckRestartSafe_AllowMinimumQuorum_Fail(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 149534)
+	skip.UnderDuress(t)
 
 	ctx := context.Background()
 	var err error
@@ -512,7 +512,7 @@ func TestCheckRestartSafe_Integration(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 150811)
+	skip.UnderDuress(t)
 
 	ctx := context.Background()
 	var err error

